packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -633,7 +636,8 @@ class _CupertinoBackGestureController<T> {
   /// The drag gesture has changed by [fractionalDelta]. The total range of the
   /// drag should be 0.0 to 1.0.
   void dragUpdate(double delta) {
-    controller.value -= delta;
+    final double newValue = controller.value - delta;
```

Comment:
I don't like this solution. Ideally the animation should be smart enough to not send animation status update when there is an ongoing gesture. However, I can't think of a good way to do it because the animation is provided from TransitionRoute and we also need to send animation status update when the gesture is lifted.

Reply:
So the clamp prevents the animation from finishing by keeping its value between 0.0 and 1.0?  I can't think of a better approach off the top of my head...